moby/moby
# Windows: Run containerd as managed process

**- What I did**
Continued from where #42089 left so now users can switch to containerd by just adding
```json
{
  "default-runtime": "io.containerd.runhcs.v1"
}
```
to `daemon.config` without need separately install or configure containerd.

Needs that containerd binaries are included to static binaries installation package #43892

Also need to be backported to 22.06.x if this approach is accepted.

**- How I did it**
Copied logic from Linux implementation and updated. In additionally used non-default named pipe so it is possible to that both Docker with containerd and separate containerd instance on same machine.

**- How to verify it**
CI was modified on way that it run Windows 2019 + containerd tests like before and Windows 2022 + containerd tests using this new approach.

Additionally manual tests can be done by
1. Add containerd binaries to same folder with Docker binaries.
2. Add daemon.json configuration
3. Start with `dockerd.exe -D` and check that you see `Starting daemon managed containerd` message.

**- Description for the changelog**
 

**- A picture of a cute animal (not mandatory but encouraged)**
